packages/aws-cdk/lib/api/aws-auth/sdk.ts

```diff
@@ -158,8 +158,8 @@ export class SDK implements ISDK {
         ...this.sdkOptions.assumeRoleCredentialsSourceDescription
           ? [`using ${this.sdkOptions.assumeRoleCredentialsSourceDescription}`]
           : [],
-        '(did you bootstrap the environment with the right \'--trust\'s?):',
         e.message,
+        '. Please make sure that this role exists in the account. If it doesn\'t exist, re-bootstrap the environment with the right \'--trust\'',
```

Comment:
@rix0rrr Given the version of the deployed bootstrap stack and the exact reason for the failure (missing role), we could display a more targeted error message. But since this may not always be possible, I updated the message slightly to make it more generic. Let me know what you think. We can dive deep on this, if you want.

Reply:
Maybe also put in there that they should use the newest version of the CDK CLI to (re)bootstrap.